tests/api/api/test_submit.py

```diff
@@ -305,6 +306,67 @@ def test_submit_job_service_accounts(
     mlconf.function.spec.service_account.default = None
 
 
+class _MockDataItem:
+    def as_df(self):
+        return pd.DataFrame({"key1": [0], "key2": [1]})
+
+
+def test_submit_job_with_hyper_params_file(
+    db: Session,
+    client: TestClient,
+    pod_create_mock,
+    k8s_secrets_mock: K8sSecretsMock,
+    monkeypatch,
+):
+    project_name = "proj-with-hyper-params"
+    project_artifact_path = f"/{project_name}"
+    tests.api.api.utils.create_project(
+        client, project_name, artifact_path=project_artifact_path
+    )
+    function = mlrun.new_function(
+        name="test-function",
+        project=project_name,
+        tag="latest",
+        kind="job",
+        image="mlrun/mlrun",
+    )
+    # set default artifact path
+    mlconf.artifact_path = "/some-path"
+    submit_job_body = _create_submit_job_body(
+        function, project_name, with_output_path=False
+    )
+
+    # Create test-specific mocks
+    auth_info = mlrun.api.schemas.AuthInfo(username="user", data_session=access_key)
+    monkeypatch.setattr(
+        mlrun.api.utils.auth.verifier.AuthVerifier(),
+        "authenticate_request",
+        lambda *args, **kwargs: auth_info,
+    )
+    orig_get_dataitem = mlrun.MLClientCtx.get_dataitem
+    mlrun.MLClientCtx.get_dataitem = unittest.mock.Mock(return_value=_MockDataItem())
```

Comment:
I would using monkeypatch here as well as if it will fail before setting the orig_get_dataitem back it will cause other tests to fail.

Reply:
I'm using the mock's `assert_called_once_with()` method on the mock, so decided to use a `Mock` object. I guess can put the revert into a `finally` clause. Let's discuss this.

Reply:
can use `with mock.patch.object(...)` as well